### DISCUSSION TITLE
How to get all row ids before tableInstance is created?

### DISCUSSION THREAD

**Discussion by simPod** [TanStack/table#3317] [Q&A]
I'd like to have all rows expanded on initial table load. In order to do that, I have to pass all row ids into `useTable()` hook.

Is there a way how to know all generated row ids in advance? When I have some grouped rows, row ids get composed based on group hierarchy so I don't know them.

I'm able to collect them from `tableInstance.rows` and then pass it to the hook again but that causes unnecessary re-render.

```ts
// example. not sure it works actually

const [rowIdsMap, setRowIdsMaps] = React.useState<{[key: string]: boolean}>({});

const tableInstance = useTable(
    ...
    initialState: {
       expanded: rowIdsMap
    }
);

...

setRowIds(tableInstance.rows.reduce(..., {}));
```

**Comment by GuptaSiddhant** [comment]
Though I’m sure there is better way to deal with this. The one I can recommend you from top of my head would be to use “useEffect” to run after table has been initiated and expand the rows. 

No need to pass rowIdsMap as initial state since it is normally used to remember user choices (saved on local storage or server). 

Run an useEffect and call function `tableInstance. toggleAllRowsExpanded(true)`. Try to make sure the effect only run once after instance is generated. 

Do share if the approach worked for you…

**Reply by simPod** [reply]
@GuptaSiddhant TBH I use exactly this approach but I wanted to make the example more illustrative.

Though I want to avoid the re-render. It causes table to blink because it renders collapsed for a short time and then renders expanded.

**Reply by babin93** [reply]
Could we add this as an Issue to have this feature available ?
